src/backoff.ts

```diff
@@ -105,16 +109,21 @@ export async function retry(
       return result;
     } catch (err) {
       originalError = err;
-      if (!predicate || !predicate(err, args)) {
-        throw err;
+      if (!predicate || !predicate(originalError, args)) {
+        throw originalError;
       }
     }
 
     numRetries++;
     if (numRetries > maxRetries) {
       // The last error that was raised before the max retries is reached
-      // will be the one thrown to the end-user.
-      throw originalError;
+      // will be the one thrown to the end-user, unless it's a "known" error
+      // that was configured for backoff.
+      if (!predicate || !predicate(originalError, args)) {
+        throw originalError;
```

Comment:
I don't understand this block. It should already be handled by the `if` on line 112 no?